src/annotator/integrations/test/html-side-by-side-test.js

```diff
@@ -163,6 +163,40 @@ the fighting was.`;
       );
     });
 
+    it('ignores fixed-position content when choosing a scroll anchor', () => {
+      // Set up a DOM structure that emulates a page with a sticky heading:
+      //
+      // <div> // Scroll root
+      //   <div> // Inner container
+      //     <nav> // Fixed-position navbar
+      //     <p>..</p> // Content
+      //   </div>
+      // </div>
+      //
+      // Here the `<nav>` contains the top-left most text node in the viewport,
+      // but we should it because of its fixed position.
+      //
+      // The inner container is used to check that the element filtering is
+      // applied as the DOM tree is recursively traversed.
```

Comment:
In an initial version of this code there was a bug where the recursive call to `textNodesInRect` did not pass the new `shouldVisit` argument. The inner `<div>` here catches this mistake.